File: common/src/main/java/io/felipeandrade/passinggas/entity/PotionEntityOfCleansing.java
package io.felipeandrade.passinggas.entity;


//
//import java.util.List;
//
//public class PotionEntityOfCleansing extends PotionEntity {
//
//    public PotionEntityOfCleansing(World world, LivingEntity owner) {
//        super(world, owner);
//    }
//
//    protected void onCollision(HitResult hitResult) {
//        super.onCollision(hitResult);
//
//        if (hitResult.getType() == HitResult.Type.ENTITY) {
//            Entity entity = ((EntityHitResult) hitResult).getEntity();
//
//            if (entity.isLiving()) {
//                ((LivingEntity) entity).clearStatusEffects();
//            }
//        }
//
//        clearEffectArea();
//    }
//
//    private void clearEffectArea() {
//        Box box = this.getBoundingBox().expand(4.0, 2.0, 4.0);
//        List<LivingEntity> list = this.getWorld().getEntitiesByClass(
//                LivingEntity.class,
//                box,
//                (entity) -> true
//        );
//
//        for (LivingEntity livingEntity : list) {
//            double d = this.squaredDistanceTo(livingEntity);
//            if (d < 16.0) {
//                livingEntity.clearStatusEffects();
//            }
//        }
//    }
//
//}

File: common/src/main/java/io/felipeandrade/passinggas/entity/PotionEntityOfCleansing.kt
package io.felipeandrade.passinggas.entity

import io.felipeandrade.passinggas.PassingGas
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.Level
import net.minecraft.world.phys.EntityHitResult
import net.minecraft.world.phys.HitResult

class PotionEntityOfCleansing : net.minecraft.world.entity.projectile.ThrowableItemProjectile {

    constructor(entityType: EntityType<out net.minecraft.world.entity.projectile.ThrowableItemProjectile>, level: Level) : super(entityType, level)

    constructor(level: Level, owner: LivingEntity) : super(PassingGas.POTION_ENTITY_OF_CLEANSING.get(), owner, level, ItemStack(PassingGas.SPLASH_POTION_OF_CLEANSING.get()))

    override fun getDefaultItem(): Item {
        return PassingGas.SPLASH_POTION_OF_CLEANSING.get()
    }

    override fun onHit(hitResult: HitResult) {
        if (hitResult.type == HitResult.Type.ENTITY) {
            val entity = (hitResult as EntityHitResult).entity
            if (entity is LivingEntity) {
                entity.removeAllEffects()
            }
        }

        if (!level().isClientSide) {
            clearEffectArea()
            this.discard()
        }
    }

    private fun clearEffectArea() {
        val box = this.boundingBox.inflate(4.0, 2.0, 4.0)
        val list = this.level().getEntitiesOfClass(LivingEntity::class.java, box) { true }

        for (livingEntity in list) {
            val d = this.distanceToSqr(livingEntity)
            if (d < 16.0) {
                livingEntity.removeAllEffects()
            }
        }
    }
}